pallets/schemas/src/types.rs

```diff
@@ -18,3 +48,126 @@ pub struct SchemaInfo {
 	/// additional control settings for the schema
 	pub settings: SchemaSettings,
 }
+
+#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen)]
+/// A structure defining name of a schema
+pub struct SchemaName {
+	/// namespace or domain of the schema
+	pub namespace: SchemaNamespace,
+	/// name or descriptor of this schema
+	pub descriptor: SchemaDescriptor,
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen, Default)]
+/// A structure defining name of a schema
+pub struct SchemaVersionId {
+	/// the index of each item + 1 is considered as their version.
+	/// Ex: the schemaId located in `ids[2]` is for version number 3
+	pub ids: BoundedVec<SchemaId, ConstU32<MAX_NUMBER_OF_VERSIONS>>,
```

Comment:
In #1763 I also have UnlockChunks as a struct with one BoundedVec field.  @enddynayn suggests it would be better for storage to simply store the BoundedVec as a type alias, and have static fns that operate on the alias.  His feeling was that unless there's a plan to add more fields later, making this a struct with methods adds to storage overhead.  I had been persuaded, but now I'm  waffling since I feel the encapsulation is appropriate and the overhead may be negligible.  Your thoughts?

Reply:
I'm 99% sure based on how scale encoding works there is no overhead